pkg/controllers/management/eks/eks_cluster_handler.go

```diff
@@ -0,0 +1,561 @@
+	}
+
+	eksClusterConfigResource := schema.GroupVersionResource{
+		Group:    eksAPIGroup,
+		Version:  "v1",
+		Resource: "eksclusterconfigs",
+	}
+
+	eksCCDynamicClient := dynamicClient.Resource(eksClusterConfigResource)
+
+	e := &eksOperatorController{
+		clusterEngqueueAfter: wContext.Mgmt.Cluster().EnqueueAfter,
+		secretsCache:         wContext.Core.Secret().Cache(),
+		templateLister:       mgmtCtx.Management.CatalogTemplates("").Controller().Lister(),
+		projectLister:        mgmtCtx.Management.Projects("").Controller().Lister(),
+		appLister:            mgmtCtx.Project.Apps("").Controller().Lister(),
+		appClient:            mgmtCtx.Project.Apps(""),
+		nsClient:             mgmtCtx.Core.Namespaces(""),
+		clusterClient:        mgmtCtx.Management.Clusters(""),
+		systemAccountManager: systemaccount.NewManager(mgmtCtx),
+		dynamicClient:        eksCCDynamicClient,
+	}
+
+	wContext.Mgmt.Cluster().OnChange(ctx, "eks-operator-controller", e.onClusterChange)
+}
+
+func (e *eksOperatorController) onClusterChange(key string, cluster *mgmtv3.Cluster) (*mgmtv3.Cluster, error) {
+	if cluster == nil || cluster.DeletionTimestamp != nil {
+		return nil, nil
+	}
+
+	if cluster.Spec.EKSConfig == nil {
+		return cluster, nil
+	}
+
+	if err := e.deployEKSOperator(); err != nil {
+		return cluster, err
+	}
+
+	if cluster.Status.Driver == "" {
+		cluster = cluster.DeepCopy()
+		cluster.Status.Driver = "EKS"
+		var err error
+		cluster, err = e.clusterClient.Update(cluster)
+		if err != nil {
+			return cluster, err
+		}
+	}
+
+	eksClusterConfigDynamic, err := e.dynamicClient.Namespace(globalNS).Get(context.TODO(), cluster.Name, v1.GetOptions{})
+	if err != nil {
+		if !errors.IsNotFound(err) {
+			return cluster, err
+		}
+		// EKS cluster config still needs to be created
+		eksClusterConfig := eksv1.EKSClusterConfig{
+			TypeMeta: v1.TypeMeta{
+				Kind:       "EKSClusterConfig",
+				APIVersion: eksV1,
+			},
+			ObjectMeta: v1.ObjectMeta{
+				Name: cluster.Name,
+			},
+			Spec: *cluster.Spec.EKSConfig,
+		}
+
+		// if cluster is imported, the EKS cluster will not be deleted on removal
+		if !aws.BoolValue(cluster.Spec.EKSConfig.Imported) {
+			eksClusterConfig.ObjectMeta.OwnerReferences = []v1.OwnerReference{
+				{
+					Kind:       cluster.Kind,
+					APIVersion: managementV3,
+					Name:       cluster.Name,
+					UID:        cluster.UID,
+				},
+			}
+		}
+
+		// convert EKS cluster config into unstructured object so it can be used with dynamic client
+		eksClusterConfigMap, err := runtime.DefaultUnstructuredConverter.ToUnstructured(&eksClusterConfig)
+		if err != nil {
+			return cluster, err
+		}
+		eksClusterConfigDynamic = &unstructured.Unstructured{
+			Object: eksClusterConfigMap,
+		}
+		eksClusterConfigDynamic, err = e.dynamicClient.Namespace(globalNS).Create(context.TODO(), eksClusterConfigDynamic, v1.CreateOptions{})
+		if err != nil {
+			return cluster, err
+		}
```

Comment:
Confused about this block, we create here and then update below ?

Reply:
The EKS Cluster Config is created and then the rancher cluster is updated with the applied spec. It will then end up in the default case and be reenqueued.